core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -596,6 +599,26 @@ object SparkSubmit extends CommandLineUtils {
       }
     }
 
+    if (isKubernetesCluster) {
```

Comment:
We should throw an error for now if in client mode with Kubernetes. There's still open questions about how we can manage the networking there, so we can revisit the support later.